src/main.cpp

```diff
@@ -51,6 +52,8 @@
 #include <sys/stat.h>
 #include <unistd.h>
 
+SignalHandler* SignalHandler::m_singleton = nullptr;
```

Comment:
Can we use a static variable in SignalHandler::instance() instead? Then the SignalHandler code can handle its own setup.